inbm/dispatcher-agent/dispatcher/aota/application_command.py

```diff
@@ -212,13 +246,16 @@ def __init__(self,  dispatcher_broker: DispatcherBroker,
         super().__init__(dispatcher_broker, parsed_manifest, dbs, update_logger=update_logger)
 
     def update(self):  # pragma: no cover
-        super().update()
+        super().update()        
         application_repo = self._download_package()
-        install_cmd = application_repo.get_repo_path() + "/" + self.resource if self.resource else ""
-        if ' ' in install_cmd or install_cmd.isspace():
-            logger.debug(f"INSTALL : {install_cmd}")
-            raise AotaError(f"File path cannot contain spaces - {install_cmd}")
-        base_command = f"/usr/bin/dpkg -i {install_cmd}"
+        path_to_pkg = application_repo.get_repo_path() + "/" + self.resource if self.resource else ""
+        if ' ' in path_to_pkg or path_to_pkg.isspace():
+            logger.debug(f"INSTALL : {path_to_pkg}")
+            raise AotaError(f"File path cannot contain spaces - {path_to_pkg}")
```

Comment:
**Suggestion:** Instead of checking for spaces in the file path after constructing it, validate self.resource before using it to prevent invalid paths. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if self.resource and (' ' in self.resource or self.resource.isspace()):
            logger.debug(f"Invalid resource name: {self.resource}")
            raise AotaError(f"Resource name cannot contain spaces - {self.resource}")
        path_to_pkg = os.path.join(application_repo.get_repo_path(), self.resource) if self.resource else ""
```